### PR TITLE
java: allow deactivation of Maven API fetching

### PR DESCRIPTION
**Pull request by BradLugo**
## Description

There's logic in the Java package scanner to disable reaching out to the internet, but we currently set a default API endpoint during `Configure` to prevent that condition. These changes introduce a `DisableAPI` flag to allow the deactivation of the fetching from the public Maven API.

### CODE REVIEW

**Comment by BradLugo** [comment]
I think I found something interesting when working on the unit tests for this PR; it seems that `TestScan` for the java package scanner tests instances where `Configure` is never called, which means we've always tested the case where the API is disabled anyway. However, maybe this analysis is incorrect (or maybe even irrelevant if the java package scanner's `Configure` is never called)

File: java/packagescanner_test.go [hunk 1]
```diff
@@ -10,6 +15,35 @@ import (
 	"github.com/quay/claircore/test"
 )
 
+// TestDisableAPI runs the java scanner over some layers known to have java
+// packages installed.
+func TestDisableAPI(t *testing.T) {
```
Comment (RTann) [inline java/packagescanner_test.go:20]: what information did we find with this test?

**Review by RTann — COMMENTED** [review]
looks good to me just one question

File: java/packagescanner_test.go (hunk 1 above)
Comment (BradLugo) [reply java/packagescanner_test.go:20]: Sorry, can you rephrase the question?

(I elaborated a bit here but I'm not sure it answers your question: https://github.com/quay/claircore/pull/1158#issuecomment-1920237990)

File: java/packagescanner_test.go [hunk 2]
```diff
@@ -10,6 +15,35 @@ import (
...
+		t.Error("http request called")
+	}))
+
+	ctx := context.Background()
+	ctx = zlog.Test(ctx, t)
+	scanner := &java.Scanner{}
+	var buf bytes.Buffer
+	cfg := &java.ScannerConfig{
+		DisableAPI:        true,
+		API:               "",
```
Comment (crozzy) [inline java/packagescanner_test.go:31]: Should this be `API: srv.URL`? Otherwise we never trigger the expected test failure (we just trigger the results parity failure).

File: java/packagescanner_test.go [hunk 3]
```diff
@@ -10,6 +15,35 @@ import (
 	"github.com/quay/claircore/test"
 )
 
+// TestDisableAPI runs the java scanner over some layers known to have java
+// packages installed.
+func TestDisableAPI(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t.Error("http request called")
```
Comment (crozzy) [inline java/packagescanner_test.go:22]: nit: maybe be kind to a future operator by explaining why this is an error

Comment (crozzy) [reply java/packagescanner_test.go:22]: Also, this should be `t.Fatal()` if the API is ever called, we don't care about results parity after this error.

**Review by crozzy — CHANGES_REQUESTED** [review]
Just a couple of comments on the test, the changes look fine to me

File: java/packagescanner_test.go (hunk 2 above)
Comment (BradLugo) [reply java/packagescanner_test.go:31]: 🤦 Thank you

**Comment by BradLugo** [comment]
Thanks for the feedback @crozzy!

File: java/packagescanner_test.go (hunk 3 above)
Comment (BradLugo) [reply java/packagescanner_test.go]: re: explaining the error, lmk if the new changes are still unclear. Definitely want to value clearer messages in the tests

**Review by crozzy — APPROVED** [review]
LGTM, thanks @BradLugo

**Review by RTann — APPROVED** [review]
